### PR TITLE
Streamline UI and allow embedding in iframes

### LINKED ISSUE #3470: Tracing View: small design tweaks
I’d like to suggest some small changes, please feel free to comment:

- [x] get rid of the yellow on-hover border for viewports
- [x] make the red/blue/green borders half as thick
- [x] make the red/blue/green less aggressive colors (I suggest rgb(200, 20, 20), (20, 200, 20), (20, 20, 200))
- [x] adjust the padding to center the text on the 3D viewport buttons 
- [x] dataset extent should not say »−infinity« while not loaded, can we just hide it?
- [x] dataset extent uses “x” instead of “×” (while resolution does this correctly)
- ~dataset extent should also say μm³ (instead of three times μm)~
- [x] remove “click this button to” in hover tooltips (should just say “copy position to clipboard”)
- ~ copy to clipboard button should use clipboard icon like the one on the auth token page~
- ~adjust overflow/padding so that the g in the “segmentation” tab label is not clipped~

### LINKED ISSUE #3521: Try to maximize area for rendering data
The end of the spectrum would be a complete full-screen data viewer.

### LINKED ISSUE #3581: Optimize/trim down WK (within iframe)
Embedding via an iframe should already work right now. We could have another flag (isIframe, lightweight=true or similar) which would make the UI more lightweight.

### LINKED ISSUE #3891: Having a small screen width causes the action bar to be rendered wrong
Screenshot from 2019-03-13 14-49-54

### PR DESCRIPTION
**Pull request by philippotto**
tldr:
- allows to embed wk within an iframe
  - logging in does not work due to third-party cookies, but there will be a warning stating this
  - the golden layout config will not be persisted within an iframe and the default layout is switched to a two-column layout
- the wk-core "toolbar" was merged into the navbar to save vertical space and make the overall UI leaner
- some stylistic tweaks here and there

### URL of deployed dev instance (used for testing):
- https://embeddablewk.webknossos.xyz

### Steps to test:
- mainly play around with screen sizes and check basic functionality

### Issues:
- fixes #3581 
- fixes #3891
- fixes #3521 imo 
- fixes #3470

### Screenshots
image
image
For small devices:
image

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- ~[ ] Updated [documentation](../blob/master/docs) if applicable~ (I created the follow-up issue #3912 for that)
- [X] Ready for review

### CODE REVIEW

**Comment by philippotto** [comment]
@daniel-wer Maybe you can have a look at https://www.playframework.com/documentation/ja/2.3.x/SecurityHeaders to see whether we should keep some of these security measures?

File: frontend/stylesheets/trace_view/_tracing_view.less [hunk 1]
```diff
@@ -168,3 +169,65 @@
...
+  .ant-radio-button-wrapper:first-child {
+    border-left: 1px solid @dark-border;
+  }
+
+  .ant-radio-button-wrapper-checked,
+  .ant-radio-button-wrapper-checked:first-child {
+    border-color: @primary-border;
+    color: @primary-border;
+  }
+
```
Comment (daniel-wer) [inline frontend/stylesheets/trace_view/_tracing_view.less]: Minor, but in order to get rid of the white vertical line between "Trace" and "Brush" (see screenshots), insert this rule:
```
.ant-radio-button-wrapper:not(:first-child)::before {
  background-color: @dark-border;
}
```
That's what I could find out using the dev tools at least.

File: frontend/javascripts/oxalis/controller.js [hunk 2]
```diff
@@ -140,7 +143,9 @@ class Controller extends React.PureComponent<PropsWithRouter, State> {
 
     app.vent.trigger("webknossos:ready");
     Store.dispatch(wkReadyAction());
-    this.setState({ ready: true });
+    setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/oxalis/controller.js]: Sorry, I should have added a comment. I did that now, the comment is:

```
// Give wk (sagas and bucket loading) a bit time to catch air before
// showing the UI as "ready". The goal here is to avoid that the
// UI is still freezing after the loading indicator is gone.
```

I reduced the timeout to 200 ms. It's quite arbitrary and we could also remove the timeout altogether, but from my experience wk is still a bit unresponsive after "set ready" is dispatched, so my hope is that the timeout makes the experience a bit smoother.

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/Filters.scala [hunk 3]
```diff
@@ -6,5 +6,5 @@ import play.api.http.HttpFilters
 import play.filters.headers.SecurityHeadersFilter
 
 class Filters @Inject()(securityHeadersFilter: SecurityHeadersFilter) extends HttpFilters {
-  def filters = Seq(securityHeadersFilter)
+  def filters = Seq()
```
[the remark this answers is not in the document]
Comment (philippotto) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/Filters.scala]: Sounds plausible. Did you test that variant? Also, did you have a look at the security rules in question? :)

File: frontend/javascripts/oxalis/controller.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (daniel-wer) [reply frontend/javascripts/oxalis/controller.js]: Thanks, that makes sense :)

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/Filters.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (daniel-wer) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/Filters.scala]: No I didn't test that, but I assumed the x-frame-options were the only header that prevented embeddability. I had a look at the remaining security rules, which are not terribly important, but I would like to keep them in there to (a) benefit from updated security headers for new play versions and (b) be able to set a content-security-policy in the future, see #3085 :)

Once we have a plan for how embeddability should work for wK, we can think about whitelisting specific domains using the [frame-ancestors](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors) directive in the CSP or we keep allowing wK to be embedded everywhere (with the "risk" of being vulnerable to click-jacking).

[the remark this answers is not in the document]
Comment (philippotto) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/Filters.scala]: Ah, now I understand. `frameOptions = null` works beautifully! Thank you!

File: frontend/javascripts/oxalis/view/action-bar/save_button.js [hunk 4]
```diff
@@ -72,10 +72,10 @@ class SaveButton extends React.PureComponent<Props, State> {
         className={this.props.className}
       >
         {this.shouldShowProgress() ? (
-          <React.Fragment>
-            {Math.floor((progressInfo.processedActionCount / progressInfo.totalActionCount) * 100)}{" "}
+          <span style={{ marginLeft: 8 }}>
+            >{Math.floor((progressInfo.processedActionCount / progressInfo.totalActionCount) * 100)}{" "}
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/view/action-bar/save_button.js]: Did you add the `>` deliberately?

Comment (philippotto) [reply frontend/javascripts/oxalis/view/action-bar/save_button.js]: Oops, no, this wasn't deliberately. You are faster than the linter :P Thank you!

File: frontend/javascripts/navbar.js [hunk 5]
```diff
@@ -22,221 +24,340 @@ type OwnProps = {|
...
+    position: "fixed",
+    width: "100%",
+    zIndex: 1000,
+    height: navbarHeight,
+    display: "flex",
+    alignItems: "center",
+    color: "rgba(255, 255, 255, 0.67)",
+    background: "#001529",
+  };
+  // used to adjust the height in login view
```
[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/navbar.js]: The code was only moved by my PR, but in fact this code is spot-on. It's the culprit for the weird navbar height. I simply removed it and now everything works fine :+1:

**Comment by philippotto** [comment]
Thanks for your thorough feedback! I think the PR is good to go now. Will probably merge tomorrow morning.

**Comment by philippotto** [comment]
@daniel-wer If you give your official approval, I'd merge this :)
